enhancements/machine-config/mco-network-configuration.md

```diff
@@ -0,0 +1,253 @@
+---
+title: mco-network-configuration
+authors:
+  - "@bcrochet"
+reviewers:
+  - "@cwalters"
+approvers:
+  - TBD
+creation-date: 2020-06-25
+last-updated: 2020-06-25
+status: provisional
+---
+
+# Extend MCO to support declarative host network configuration
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The ability to configure additional host networking parameters is paramount on some platforms, especially baremetal. Baremetal nodes usually have many interfaces that need to be dynamically configured to access other, critical datacenter resources. Doing so with MachineConfig would incur reboots that are expensive in time on the baremetal platform. The ability to configure host networking parameters without rebooting is desirable. It is also desirable to rollback automatically on failure, as opposed to failing on reboot.
```

Comment:
Is there explicit mention of what kind of networking parameters will be supported in this proposal?
There are several other OLM managed components such as PTP/SR-IOV Operators rely on the network configuration or has the ability to manipulate the network device parameters in a declarative way, my concern and question is how those components should interact and coordinate with this proposed change? should this be captured in the proposal?

Reply:
Replacing or interacting with any other operator is a non-goal.

Reply:
@bcrochet thanks, I understand this may not be the goal of this proposal, but would you see this as a potential issue if there are multiple components on the platform that try to manipulate the same device's network configuration or some other components that rely on the same network device to function properly? do you think we shall have another proposal to address this issue or maybe list it as a risk that may break other functions?

Reply:
I didn't see the risk of 'multiple components may manipulate the same network device with declarative APIs' is mentioned in the proposal, why it is resolved?